components/butterfly/src/network.rs

```diff
@@ -134,6 +132,19 @@ impl GossipPullSocket for ZmqGossipSocket {
     }
 }
 
+/// This is a wrapper to provide interior mutability of an underlying `zmq::Context` and allows
+/// for sharing/sending of a `zmq::Context` between threads.
+struct ServerContext(UnsafeCell<zmq::Context>);
+
+impl ServerContext {
+    pub fn as_mut(&self) -> &mut zmq::Context {
+        unsafe { &mut *self.0.get() }
```

Comment:
Why is the `unsafe` needed here? Should it be documented in some way?

Reply:
No idea, I didn't write it - I moved this code to the only file where it's used.